include/picongpu/param/fieldSolver.param

```diff
@@ -60,12 +60,16 @@ namespace fields
     /** FieldSolver
      *
      * Field Solver Selection:
-     *  - Yee< CurrentInterpolation > : standard Yee solver
+     *  - Yee< CurrentInterpolation > : standard 2nd order Yee solver
      *  - YeePML< CurrentInterpolation >: standard Yee solver using Perfectly Matched Layer Absorbing Boundary Conditions (PML)
      *  - Lehe< CurrentInterpolation >: Num. Cherenkov free field solver in a chosen direction
      *  - LehePML< CurrentInterpolation >: Num. Cherenkov free field solver in a chosen direction
      *                                     using Perfectly Matched Layer Absorbing Boundary Conditions (PML)
      *  - DirSplitting< CurrentInterpolation >: Sentoku's Directional Splitting Method
+     *  - ArbitraryOrderFDTD< CurrentInterpolation, 8 >: 16th order solver using 8 neighbors for the calculation of spatial derivatives.
```

Comment:
Do you think it's more easy for a user to operate with order or number of neighbors? I would think the order, and then we could internally derive the number of neighbors from it.

Reply:
An unrelated note: this order is just for spatial derivatives. So this solver is FDTD(2, 16) from the literature. Perhaps we could clarify it more in the comment.

Reply:
I totally agree to your second comment.

Regarding the first, I chose number of neighbors, since users could be tempted to choose uneven numbers for the order which is not valid as the order is always twice the number of neighbors, i.e. 2n. Therefore an extra check with a corresponding error message would be necessary, when taking the order as input, which I wanted to avoid due to laziness :grin:

Reply:
I see your motivation. But feel here the order is a user-side parameter, while the number of neighbors needed is a purely computational one, and PIConGPU already automatically takes care of it via margins, so that technically a user is not required to know how order corresponds to the number of neighbors.

Reply:
On the contrary, I am convinced the user should know this connection. It is essential for a users to know the underlying principle of the used methods. This is also the reason why I added this larger amount of documentation.
From a user perspective, I would argue that it does not matter whether I give order or number of neighbors.

As a side effect, the user may be able to e.g. relate the compile error `Entry function '...' uses too much shared data` when using too many neighbors to its cause.